hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/management/actionhistory/ActionHistoryLayout.java

```diff
@@ -0,0 +1,212 @@
+/**
+ * Copyright (c) 2015 Bosch Software Innovations GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.hawkbit.ui.management.actionhistory;
+
+import org.eclipse.hawkbit.repository.DeploymentManagement;
+import org.eclipse.hawkbit.repository.model.Target;
+import org.eclipse.hawkbit.ui.common.grid.AbstractGrid;
+import org.eclipse.hawkbit.ui.common.grid.AbstractGridComponentLayout;
+import org.eclipse.hawkbit.ui.common.grid.DefaultGridHeader;
+import org.eclipse.hawkbit.ui.common.grid.DefaultGridHeader.AbstractHeaderMaximizeSupport;
+import org.eclipse.hawkbit.ui.common.table.BaseEntityEventType;
+import org.eclipse.hawkbit.ui.management.event.ManagementUIEvent;
+import org.eclipse.hawkbit.ui.management.event.TargetTableEvent;
+import org.eclipse.hawkbit.ui.management.state.ManagementUIState;
+import org.eclipse.hawkbit.ui.utils.HawkbitCommonUtil;
+import org.eclipse.hawkbit.ui.utils.SPUIDefinitions;
+import org.eclipse.hawkbit.ui.utils.UINotification;
+import org.eclipse.hawkbit.ui.utils.VaadinMessageSource;
+import org.vaadin.spring.events.EventBus.UIEventBus;
+import org.vaadin.spring.events.EventScope;
+import org.vaadin.spring.events.annotation.EventBusListenerMethod;
+
+import com.vaadin.ui.UI;
+
+/**
+ * Layout responsible for action-history-grid and the corresponding header.
+ */
+public class ActionHistoryLayout extends AbstractGridComponentLayout {
+    private static final long serialVersionUID = -3766179797384539821L;
+
+    protected final transient DeploymentManagement deploymentManagement;
+    protected final UINotification notification;
+    protected final ManagementUIState managementUIState;
+
+    protected transient AbstractGrid<?>.DetailsSupport details;
+    protected Long masterForDetails;
+
+    /**
+     * Constructor.
+     *
+     * @param i18n
+     * @param deploymentManagement
+     * @param eventBus
+     * @param notification
+     * @param managementUIState
+     */
+    public ActionHistoryLayout(final VaadinMessageSource i18n, final DeploymentManagement deploymentManagement,
+            final UIEventBus eventBus, final UINotification notification, final ManagementUIState managementUIState) {
+        super(i18n, eventBus);
+        this.deploymentManagement = deploymentManagement;
+        this.notification = notification;
+        this.managementUIState = managementUIState;
+        init();
```

Comment:
Constructor should not call overridables see: http://www.javapractices.com/topic/TopicAction.do?Id=215